Findjump

ssiegelx opened the pull request:
The `find_jump_analyzer` takes about 2 hours to process 24 hours worth of data on marimba.

nritsche commented on dias/analyzers/find_jump_analyzer.py:
```diff
@@ -0,0 +1,1230 @@
+        Loop over frequencies and inputs and find jumps.
+        Write locations and snapshots of the jumps to disk.
+        """
+        self.run_start_time = time.time()
```
Why is run_start_time a class member?

nritsche commented on dias/analyzers/find_jump_analyzer.py:
```diff
@@ -0,0 +1,1230 @@
+                for ff, freq in enumerate(dfreq):
+
+                    print_cnt = -1
+                    msg = ("Processing frequency %d (%0.2f MHz, %0.2f MHz)" %
```
It wouldn't be clear from looking at the log, that the first number is the centre freq and the second is the width.

ssiegelx commented on conf/tasks/find_jump.conf:
```diff
@@ -0,0 +1,34 @@
+analyzer: "dias.analyzers.FindJumpAnalyzer"
+archive_data_dir: "/mnt/gong/not_archived"
```
We only archive `chimecal` files near the transit of the four brightest point sources.

For this analysis, we need the `chimecal` data with full time coverage, so we have to use the files in`not_archived`.  These are typically available for 1 - 2 months.

This means that this analyzer cannot use Finder and has to do its own indexing of the `not_archived` files.

nritsche replied:
Can it not use Finder with a different path?

ssiegelx replied:
No I asked @ketiltrout about this and apparently alpenhorn does not index the files in `not_archived`.

ssiegelx replied:
I've also tried using Finder with a different path and it returns nothing.

nritsche commented on dias/analyzers/find_jump_analyzer.py:
```diff
@@ -0,0 +1,1230 @@
+                                        "per frequency.")
+
+        # Determine default achive attributes
+        tag = subprocess.check_output(["git", "-C", os.path.dirname(__file__),
```
This will not work after installation. Possible fix is like in #79

nritsche commented on dias/analyzers/find_jump_analyzer.py:
```diff
@@ -0,0 +1,1230 @@
+        # Determine the range of time to process
+        end_time = datetime.datetime.utcnow() - self.offset
+
+        cursor = self.data_index.cursor()
```
This needs to be fixed, see #80 #81, #82

ssiegelx commented:
> Great!
> Should the timing logs be on debug level?
> It generally looks like it might produce a lot of log.

I assume that debug level will not be printed to the log in production mode?   I've moved several other messages from info to debug to reduce the total number of messages produced.

> About the metrics: Why not instead of all the data metrics just one metric called jumps_total with the labels `freq` and `input`? Would that be too heavy? (Maybe with only one of the labels to reduce the amount of data exported as metrics)

I don't think this would be heavy because it will only run once per day and we only process a single frequency.  It looks like there are functions that can be used in Prometheus queries to generate the relevant statistics from `jumps_total`, so I've implemented your suggestion.

nritsche commented:
> I assume that debug level will not be printed to the log in production mode?
You can configure that in the task file.

> It looks like there are functions that can be used in Prometheus queries to generate the relevant statistics from `jumps_total`, so I've implemented your suggestion.
Great! Can you have a look at the merge conflict?

ssiegelx commented:
[DocLib 788](https://bao.chimenet.ca/doc/documents/788) describes this analyzer and the associated theremin graph.

Note that I've converted past data products that were generated using an initial version of this code to the format output by Dias and moved them to `/mnt/gong/dias/data/find_jump` so that they can also be viewed with the Theremin graph.

nritsche commented:
Looks good to me. @ketiltrout @markhalpern
